src/output-json-dnp3.c

```diff
@@ -0,0 +1,464 @@
+/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "debug.h"
+#include "detect.h"
+#include "pkt-var.h"
+#include "conf.h"
+
+#include "threads.h"
+#include "threadvars.h"
+#include "tm-threads.h"
+
+#include "util-print.h"
+#include "util-unittest.h"
+#include "util-buffer.h"
+#include "util-crypt.h"
+#include "util-debug.h"
+
+#include "app-layer.h"
+#include "app-layer-parser.h"
+#include "app-layer-dnp3.h"
+#include "app-layer-dnp3-objects.h"
+
+#include "detect-dnp3.h"
+
+#include "output.h"
+#include "output-json.h"
+#include "output-json-dnp3-objects.h"
+
+#ifdef HAVE_LIBJANSSON
+#include <jansson.h>
+
+typedef struct LogDNP3FileCtx_ {
+    LogFileCtx *file_ctx;
+    uint32_t    flags;
+    uint8_t     include_object_data;
+} LogDNP3FileCtx;
+
+typedef struct LogDNP3LogThread_ {
+    LogDNP3FileCtx *dnp3log_ctx;
+    uint32_t        count;
+    MemBuffer      *buffer;
+} LogDNP3LogThread;
+
+static int JsonDNP3PrefixIsSize(uint8_t prefix_code)
```

Comment:
any issue just putting it into the app-layer for now, as both pull in app-layer-dnp3.h?